pulsar-client-go/pulsar/c_producer.go

```diff
@@ -298,6 +298,7 @@ func pulsarProducerSendCallbackProxyWithMsgID(res C.pulsar_result, messageId *C.
 		sendCallback.callback(getMessageId(messageId), newError(res, "Failed to send message"))
 	} else {
 		sendCallback.callback(getMessageId(messageId), nil)
+		C.pulsar_message_id_free(messageId)
```

Comment:
@bschofield I think this free should not be done, because `getMessageId(messageId)` is already attaching a finalizer to the Go object and it will free the message id automatically:

```
func getMessageId(messageId *C.pulsar_message_id_t) MessageID {
	msgId := &messageID{ptr: messageId}
	runtime.SetFinalizer(msgId, messageIdFinalizer)
	return msgId
}
```

Reply:
You're absolutely right. I didn't even think to check that getter for side effects. Removed in latest commit. Thanks!